### ISSUE TITLE
Training datasets

### ISSUE THREAD

**Issue by AndreyGalitsin** [UttaranB127/take_an_emotion_walk#2]
Hi, thank you for your work! 
I have 2 questions about training process and training datasets. Can you explain me please:

1. You have a mention of 2 different datasets in your codes: **'ewalk'** and **'edin'.** Also you say, that you use the Emotion-Gait dataset for this work (**data_edin_locomotion_pose_diff_aff_drop_5.npz**). So is the Emotion-Gait dataset the same as **'edin'** dataset in your meaning?
2. What is the second dataset **'ewalk'** (with 16 keypoints)? Is it absolutely another dataset or it is possible to create it from **'edin'** and train your model on it? (I want to use this 3d pose detector in demo [lightweight-human-pose-estimation-3d-demo.pytorch](https://github.com/Daniil-Osokin/lightweight-human-pose-estimation-3d-demo.pytorch) but has only 16 output points, so I can't use it for your trained model required 21 pose points)

I will wait for your answer.
All the Best!

**Comment by UttaranB127** [comment]
Hi Andrey,

The Emotion-Gait dataset is a mixture of both 'ewalk' and 'edin'. 'ewalk' is a smaller, older dataset that we collected in-house and is completely separate from the 'edin' dataset. The trained network weights that we provide are for the 21-joint 'edin' dataset. You can retrain our network with the 16-joint 'ewalk' dataset that we also provide. However, it only has 342 samples, so may not be best suited for a deep network. You can also consider reducing the 21-joint samples of the 'edin' dataset to 16 joints (maybe eliminate 5 joints in total from the spine, arms, and legs that you don't need) and retrain our network with those samples.

Hope this answers your question!

**Comment by AndreyGalitsin** [comment]
Thank you, it’s good idea, I will try to retrain it. Tell me please, am I right, I have to change only joint_parents = np.array([-1, 0, 1, 2, 3, 0, 5, 6, 7, 0, 9, 10, 11, 11, 13, 14, 15, 11, 17, 18, 19])? 
Can you also explain me please, how did you form this array? (I have read another issue, but it’s not still clear for me)
For example, how joint_parents array should loos like if I have only 6 points(head, neck, 2 shoulders and 2 arms)?
Also can you explain please , what does num_labels = [4, 3, 3] mean?

**Comment by UttaranB127** [comment]
Yes, changing the parents should work. It might be better to remove the unused joints altogether from the joint positions array and the joint parents array to avoid unexpected bugs.

The joint parents array is created using the formula `joint_parents[i] = parent(i)`, where `i` is the joint index, and `parent(i)` is the parent of the joint `i`. The parent is decided based on the joint structure. We consider the pose graph to be a directed graph, with the root being the source (hence has no parent, written as `parent(0) = -1`), and the extremities (head, hands, feet) being the leaves.

If you consider only those 6 joints, one possible assignment is to set neck=0 (the root joint), head=1, l_shoulder=2, l_arm=3, r_shoulder=4, r_arm=5. Then the joint parents array will be `[parent(neck), parent(head), parent(l_shoulder), parent(l_arm), parent(r_shoulder), parent(r_arm)] = [-1, 0, 0, 2, 0, 4]`.

For the labels, we originally considered classes for three kinds of affect: emotion, behavior, and personality. Emotion had 4 classes, behavior had 3, and personality had 3. In the paper, we only used emotions, so only the first 4 are relevant in the code.

**Comment by AndreyGalitsin** [comment]
Thank you very much, I understand!
